ons-spark/spark-analysis/big-data-workflow.md

```diff
@@ -0,0 +1,118 @@
+# **Big data workflow**
+Working with big data efficiently presents a major challenge to analysts. In particular, working with large datasets can use large amounts of computational resource, which can be very expensive. Inefficient code or working practices can unnecessarily increase the costs of running your analysis, as well as the time it takes you to run your code. However, there are good practices that can be employed to help minimise costs without sacrificing the quality of your analysis. We have produced this page as a general guide to best practice in working with big data, including tips for general workflow and managing pipeline development, working with samples and subsets of your data, and optimising analysis with Spark.
+
+
+## **General workflow and project planning**
+
+### 1. Plan your analysis
+- A simplified version of the process of developing a new pipeline might look something like this:
+
+
+```{figure} ../images/big_data_pipeline_stages.png
+---
+width: 100%
+name: BigDataPipelineStages
+alt: Diagram showing potentials steps in the process of developing a new big data pipeline.
+---
+Fig 1. Big data pipeline development stages
+```
+
+- Splitting your development into clearly defined stages like this will help with applying some of the tips we have below for keeping your resource use to a minimum. For example, working with small samples or subsets of your data in the EDA phase.
+
+### 2. [Version control](https://gitlab-app-l-01/DAP_CATS/guidance/-/wikis/Git-and-GitLab) your code using Git and apply [RAP good practice](https://best-practice-and-impact.github.io/qa-of-code-guidance/intro.html) 
+
+- This will make it easier to track what has and hasn't already been done and for others to understand what your code is doing.
+
+- It is a good idea to agree on a coding plan including code style for the project and Git branch strategies in advance.
+
+- Ensure code is readable, well documented, and follows a logical structure.
+
+### 3. Use [code reviews](https://best-practice-and-impact.github.io/qa-of-code-guidance/peer_review.html) 
+- Peer review of code should be used to quality assure and improve your code. A fresh pair of eyes may also be able to spot places where things could be done more efficiently to save computing resource.
+
+- This also helps other members of your team understand the code in detail and can help build capability in the team. It is important to avoid situations where a single developer is relied upon to understand, write and maintain all code as this can cause major problems if that person is away or leaves the team. 
+
+## **Data exploration and quality**
+
+### 4. Be aware of the size of your dataset
+- How many rows and columns does your dataset have?
+- If the dataset is relatively small (less than a few million rows) use Python or R for analysis instead of PySpark or SparklyR.
```

Comment:
You could link to the when to use Spark page of the book here for some extra info about when Spark is appropriate.

Reply:
Added